schematic/__init__.py

```diff
@@ -159,11 +207,48 @@ def set_up_tracing(session: requests.Session) -> None:
         attribute_propagator = AttributePropagatingSpanProcessor(["user.id"])
         trace.get_tracer_provider().add_span_processor(attribute_propagator)
         exporter = OTLPSpanExporter(session=session)
+        # Overwrite the _readable_span method to redact sensitive data
+        SpanSdk._readable_span = _readable_span_alternate
         trace.get_tracer_provider().add_span_processor(BatchSpanProcessor(exporter))
     else:
         trace.set_tracer_provider(TracerProvider(sampler=ALWAYS_OFF))
 
 
+original_function_readable_span = SpanSdk._readable_span
+
+
+def _readable_span_alternate(self: SpanSdk) -> ReadableSpan:
+    """Alternative function to the readable span. This function redacts sensitive data from the span attributes and events.
+
+    Args:
+        self (SpanSdk): _readable_span method of the SpanSdk class
+
+    Returns:
+        ReadableSpan: a new readable span that redacts sensitive data
+    """
+    redacted_events = []
+    # remove sensitive information in span status description
+    # this avoid statusMessage on signoz contains sensitive information
+    if self._status.status_code == trace.StatusCode.ERROR:
+        status_description_redacted = redact_string(str(self._status.description))
+        self._status = Status(self.status.status_code, status_description_redacted)
```

Comment:
Same comment from `Event` applied here. Can we set `_description` on `self._status` instead of creating a new `Status` class instance.

Reply:
Yeah it worked! Thanks.